### ISSUE TITLE
Fix kube-proxy static pods to daemonset migration path

### ISSUE THREAD

**Issue by MrHohn** [kubernetes/kubernetes#65114]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug
/sig network

From https://github.com/kubernetes/kubernetes/issues/64899, it seems like kubelets is intended made to not be able to self-label anymore (kubernetes/community#911). Current kube-proxy migration path depends on kubelets being able to self-label upon node upgrade, hence https://k8s-testgrid.appspot.com/sig-network-gce#gci-gce-latest-upgrade-kube-proxy-ds is mostly failing.

One fix in my mind might be to migrate off the label logic from kubelet to somewhere else (possibly in the cluster management layer).

This will block https://github.com/kubernetes/kubernetes/issues/23225.

/assign
cc @thockin @bowei @dcbw

**Comment by liggitt** [comment]
> Current kube-proxy migration path depends on kubelets being able to self-label upon node upgrade

Self labeling only ever happened when the kubelet deleted/recreated it's node object. 

That only happened in cloud provider specific situations, it was not a general mechanism things deployed on kubelets should depend on happening on upgrade. 

The self-deletion behavior was an artifact of the external ID being immutable and required for cloud provider identification. That field has been deprecated since 1.2, and as of 1.11 is no longer populated or required by the in-tree cloud providers, so the kubelet no longer self-deletes its node object. 

Additionally, GCE stabilized the external ID to be identical before/after upgrade. 

In short, delete/recreate of the Node object is not something to depend on happening on upgrade boundaries.

**Comment by MrHohn** [comment]
Thanks for the explanation, that is informative. Relevant conversions on https://github.com/kubernetes/kubernetes/issues/59314 as well.

Don't rely on the self-delete logic seems to be the right thing to do.

**Comment by liggitt** [comment]
https://github.com/kubernetes/kubernetes/pull/74424 is relevant

**Comment by MrHohn** [comment]
A migration path may not may not be necessary to be implemented in the OSS kube-up scripts, I will close this for now.
